packages/webvh/src/anoncreds/services/WebVhAnonCredsRegistry.ts

```diff
@@ -0,0 +1,552 @@
+        )
+      }
+      agentContext.config.logger.trace(`Resource ${resourceId} issuer ID matches DID.`)
+
+      return { resourceObject, resolutionResult }
+    } catch (error) {
+      agentContext.config.logger.error(`Error in did:webvh _resolveAndValidateAttestedResource:`, error)
+      throw error
+    }
+  }
+
+  public async getSchema(agentContext: AgentContext, schemaId: string): Promise<GetSchemaReturn> {
+    try {
+      const { resourceObject, resolutionResult } = await this._resolveAndValidateAttestedResource(
+        agentContext,
+        schemaId,
+        'schema'
+      )
+
+      // --- Parsing Step (Simplified) ---
+      const schemaContent = resourceObject.content
+
+      // Type check to ensure we have a schema content object
+      if (!('attrNames' in schemaContent) || !('name' in schemaContent) || !('version' in schemaContent)) {
+        throw new CredoError(`Parsed resource content for ${schemaId} is not a valid schema.`)
+      }
+
+      // Use the issuerId from the content if available, otherwise extract from schemaId
+      // We need the hash part for this, extract it again or pass from helper?
+      // For simplicity, extract again.
+      const parts = schemaId.split(/[:/]/)
+      const expectedMultibaseHash = parts[parts.length - 1]
+      const issuerId =
+        schemaContent.issuerId ||
+        schemaId.substring(0, schemaId.lastIndexOf(expectedMultibaseHash)).replace(/[:/]$/, '')
+
+      return {
+        schema: {
+          attrNames: schemaContent.attrNames,
+          name: schemaContent.name,
+          version: schemaContent.version,
+          issuerId: issuerId,
+        },
+        schemaId,
+        resolutionMetadata: resolutionResult.dereferencingMetadata || {},
+        schemaMetadata: resolutionResult.contentMetadata || {},
+      }
+    } catch (error) {
+      agentContext.config.logger.error(`Error retrieving schema '${schemaId}' via did:webvh`, {
+        error: error instanceof Error ? error.message : String(error),
+        schemaId,
+      })
+
+      let errorType: GetSchemaReturn['resolutionMetadata']['error'] = 'internalError'
+      if (error instanceof CredoError) {
+        if (error.message.includes('parse resource data')) errorType = 'invalidJson'
+        else if (error.message.includes('resolve') || error.message.includes('missing data')) errorType = 'notFound'
+        else if (error.message.includes('attested')) errorType = 'invalid'
+        else if (error.message.includes('proof')) errorType = 'invalid'
+        else if (error.message.includes('hash mismatch')) errorType = 'invalid'
+        else if (error.message.includes('not a valid schema')) errorType = 'invalid'
+        else if (error.message.includes('Invalid schema ID')) errorType = 'invalidDid'
+        else if (error.message.includes('Issuer ID mismatch')) errorType = 'invalid'
+        else if (error.message.includes('missing a valid issuerId')) errorType = 'invalid'
+      }
```

Comment:
Where are these error messsages constructed? We don't use error messages anywhere to handle errors as it's very brittle. If you want to catch these cases, best i think is to create and throw specific errors that you can check with instanceof